sql: internal error due to JSON value too large

roachtest.sqlsmith/setup=seed/setting=no-ddl [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4455314&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4455314&tab=artifacts#/sqlsmith/setup=seed/setting=no-ddl) on master @ [978744dbb94b03497154547ce7fa97ea3be9a401](https://github.com/cockroachdb/cockroach/commits/978744dbb94b03497154547ce7fa97ea3be9a401):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/sqlsmith/setup=seed/setting=no-ddl/run_1
	sqlsmith.go:258,sqlsmith.go:297,test_runner.go:779: error: pq: internal error: JSON value too large: 281916604 bytes
		stmt:
		SELECT
			json_agg(tab_3935._jsonb)::JSONB AS col_8810, 'f0692c6f-4ec1-4917-b9af-6a7917349b84':::UUID AS col_8811
		FROM
			defaultdb.public.seed@seed__int8__float8__date_idx AS tab_3935,
			defaultdb.public.seed AS tab_3936
			JOIN defaultdb.public.seed AS tab_3937 ON (tab_3936._jsonb) = (tab_3937._jsonb)
		GROUP BY
			tab_3935._jsonb, tab_3935._uuid
		HAVING
			bool_and(tab_3935._bool::BOOL)::BOOL
		ORDER BY
			tab_3935._uuid ASC, tab_3935._uuid;
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed/setting=no-ddl.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13371

--- a/pkg/util/json/encode.go
+++ b/pkg/util/json/encode.go
@@ -37,7 +37,7 @@ const jEntryLen = 4
 // JSON value, but check it just to be safe.
 func checkLength(length int) error {
 	if length > maxByteLength {
-		return errors.AssertionFailedf("JSON value too large: %d bytes", errors.Safe(length))
+		return errors.Newf("JSON value too large: %d bytes", errors.Safe(length))
 	}
 	return nil
 }

